app_legacy/Site/Commands/UpdateUserTimestamps.php

```diff
@@ -0,0 +1,166 @@
+                'Updated' => DB::raw('Updated'),
+            ]);
+
+        // https://web.archive.org/web/20121121150009/http://retroachievements.org/
+        User::whereIn('User', [
+            'Dave',
+            'Jimmy',
+            'Batman',
+            'Steve',
+            'Michael',
+            'poip',
+            'Sonic',
+            'qwer',
+            'Bob',
+            'Dan',
+            'Jim',
+            'qweqwe',
+            'qwe',
+        ])
+            ->update([
+                'Created' => '2012-11-21 13:37:00',
+                'Updated' => DB::raw('Updated'),
+            ]);
+
+        // Teario - first achievement unlock
+        User::where('User', 'Teario')
+            ->update([
+                'Created' => '2012-12-29 13:37:00',
+                'Updated' => DB::raw('Updated'),
+            ]);
+
+        // Same as nightgoat99
+        User::where('User', 'nightgoat')
+            ->update([
+                'Created' => '2013-03-02 04:17:05',
+                'Updated' => DB::raw('Updated'),
+            ]);
+
+        $this->info('Updated initial account creation timestamps');
+    }
+
+    /**
+     * Iterate through user accounts by ID, descending, fill gaps and correct order of timestamps.
+     */
+    private function interpolateAccountCreationDates(): void
+    {
+        Eloquent::unguard();
+        // 106605, created on 2019-09-22 is the first user to have a real creation timestamp
+        $maxId = 106605;
+
+        $this->line('Fetching forum topic comment timestamps...');
+        $forumTopicTimestamps = User::withTrashed()
+            ->selectRaw('UserAccounts.ID, UserAccounts.User, UserAccounts.Created, MIN(ForumTopicComment.DateCreated) ForumTopicTimestamp')
+            ->leftJoin('ForumTopicComment', 'Author', '=', 'UserAccounts.User')
+            ->where('UserAccounts.ID', '<=', $maxId)
+            // Manually deleted accounts which have old forum topic comments and would mess with the sequence
+            ->whereNotIn('UserAccounts.ID', [
+                92596,
+                92599,
+                92600,
+                92601,
+                92602,
+            ])
+            ->whereRaw('ForumTopicComment.DateCreated < UserAccounts.Created')
+            ->groupByRaw('UserAccounts.ID, UserAccounts.User, UserAccounts.Created')
+            ->orderByDesc('UserAccounts.ID')
+            ->get()
+            ->keyBy('ID');
+        $this->info('Found ' . $forumTopicTimestamps->count() . ' earlier forum topic comment timestamps');
+
+        $this->line('Fetching comment timestamps...');
+        $commentTimestamps = User::withTrashed()
+            ->selectRaw('UserAccounts.ID, UserAccounts.User, UserAccounts.Created, MIN(Comment.Submitted) CommentTimestamp')
+            ->leftJoin('Comment', 'UserID', '=', 'UserAccounts.ID')
+            ->where('UserAccounts.ID', '<=', $maxId)
+            ->whereRaw('Comment.Submitted < UserAccounts.Created')
```

Comment:
I did a fresh import and reran the command - both statements returned an empty list for me 🤔

Reply:
I ran them on prod. First returns two records. Second returns seven.

Reply:
That sounds fine. I just realized the first queries I posted didn't handle the accounts without a created timestamp, checked for them, and found some. I didn't cross-reference with the interpolated values.

Reply:
Oh, I see! Shouldn't be an issue because from what I've seen all of those 9 accounts will receive a much earlier Created date by interpolation than their respective comments provide.

Reply:
```
| ID  | min(ftc.DateCreated) | Interpolated
|-----|----------------------|
|   5 | 2018-06-09 06:59:13  | 2012-11-21 13:37:00
|   7 | 2018-08-06 14:44:41  | 2012-11-21 13:37:00
|  13 | 2022-04-27 11:04:15  | 2012-11-21 13:37:00
|  34 | 2023-01-25 05:20:29  | 2013-04-08 05:30:23
|  82 | 2018-07-08 13:45:09  | 2013-04-09 21:42:30
|  90 | 2021-01-19 01:49:37  | 2013-04-10 23:51:54
| 265 | 2022-09-22 22:44:40  | 2013-04-15 03:02:05

| ID     | min(c.Submitted)    | Interpolated
|--------|---------------------|
|  33451 | 2022-11-20 14:21:13 | 2017-01-14 13:02:40
| 104955 | 2020-09-24 23:09:49 | 2019-09-01 20:17:24
```